assume/common/scenario_loader.py

```diff
@@ -359,8 +363,17 @@ async def load_scenario_folder_async(
             [all_operators, heatpump_units.unit_operator.unique()]
         )
 
+    def unit_operator_callback(unit_operator):
+        unit_operator.context.data_dict["price_forecast"] = world.forecast_providers[
+            "EOM"
+        ].forecasts["price_forecast"]
+        unit_operator.context.data_dict[
+            "res_demand_forecast"
+        ] = world.forecast_providers["EOM"].forecasts["residual_load_forecast"]
+
     for company_name in set(all_operators):
         world.add_unit_operator(id=str(company_name))
+        unit_operator_callback(world.unit_operators[company_name])
```

Comment:
unit_operator = world.unit_operators[company_name]
and move line 367-373 here?

Reply:
I would suggest leaving that in the callback function, since it is the preparation of the forecast role functionality.

Reply:
Yes, if that is needed for further things, thats fine